File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = {GCD(n, m), LCM(n, m)};
        return answer;
    }
    
    int GCD(int num1, int num2) {
        if (num2 == 0) return num1;
        return GCD(num2, num1 % num2);
    }
    
    int LCM(int num1, int num2) {
        return num1 * num2 / GCD(num1, num2);
    }
}

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <bits/stdc++.h>
#define fastIO cin.tie(0)->sync_with_stdio(0)
using namespace std;

int gcd(int num1, int num2) {
    if (num2 == 0) return num1;
    return gcd(num2, num1 % num2);
}
    
int lcm(int num1, int num2) {
    return num1 * num2 / gcd(num1, num2);
}
    
    
int main() {
    fastIO;
    
    int a, b;
    
    cin >> a >> b;
    
    int c = gcd(a, b);
    int d = lcm(a, b);
    
    cout << c << '\n' << d;
    
    return 0;
}